resources/assets/js/components/Application/BasicInfo/BasicInfoPage.tsx

```diff
@@ -46,6 +43,8 @@ const BasicInfoPage: React.FunctionComponent<BasicInfoPageProps> = ({
   const job = useJob(jobId);
   const steps = useJobApplicationSteps();
 
+  useTouchApplicationStep(applicationId, "basic", dispatch);
```

Comment:
I'm getting a flash of the error state. When the user hits that step for the first time, on the backend it sets the step to touched. Then the response payload has that step set to `error`. When they complete the step and move to the next, that's when you'll see a flash of the error state.  

To prevent this we can add a loading state. This would mean adding a `stepIsUpdating` state to the applications reducer, adding new action, etc.

Reply:
I see what you mean now.

I think adding a stepIsUpdating state is the right approach. In terms of what to show for loading state - maybe just replace any 'error' states with 'default' temporarily, as 'error' is the most disruptive.

Reply:
Okay, I've made the changes I just mentioned.